### DISCUSSION TITLE
Manual Deployment Methods & Instructions

### DISCUSSION THREAD

**Discussion by cvdsouza** [intelowlproject/IntelOwl#108] [General]
Hi , 
I'm looking to deploy IntelOwl manually in Google Cloud. I want to see what technologies of GCP i can leverage to make it a scalable deployment. Do you have any instructions for manual deployment ?

**Comment by eshaan7** [comment]
Hi. Thank you for your interest. IMHO, there's 2 ways to go about this.

1. If you are not a GCP and/or django expert and you want to do it fast, you could create a new VM instance, clone the [v1.0.0](https://github.com/intelowlproject/IntelOwl/releases/tag/1.0.0) release, follow the [Installation.md](https://intelowl.readthedocs.io/en/latest/Installation.html) steps and just do `docker-compose up`. The application is built to be efficient and scalable in a way that this should be enough in more cases.

2. For a more scalable and efficient deployment, you could leverage different GCP products. Here's an overview:
 - The main (a) django application along with (b) celery and (c) rabbit-mq running in docker via the `docker-compose.yml` file.
 - [Cloud SQL](https://cloud.google.com/sql) for a PostgreSQL database which is used by the main django application.
 - [Cloud Storage Buckets](https://cloud.google.com/storage/docs/creating-buckets) for static files (frontend pages), which will serve as a CDN.
- Use nginx for reverse proxies on the host VM instance. There's already configured nginx config files available under `configuration/` folder.
- And finally [Let's Encrypt Bot](https://certbot.eff.org/) for SSL. 

This is very well laid out in details on [GCP's django guide](https://cloud.google.com/python/django). 

We will leave this thread open as a _discussion_ so others could benefit from this too or better yet if someone has ideas regarding this, feel free to join this discussion.

**Reply by eshaan7** [reply]
IntelOwl on GKE: [guide](https://intelowl.readthedocs.io/en/latest/Advanced-Usage.html?highlight=gke#google-kubernetes-engine-deployment)

**Comment by Slavell** [comment]
Hi, I deployed the IntelOwl on virtual machine using the docker, i've used this product by 2 days, then i've added new keys to env_file_app, tried to rebuild the project and i'm receiving the 500 error now while trying to pass authorization... What could go wrong???

**Reply by mlodic** [reply]
Hey, we might need more detailed information to detect what went wrong (stack trace of 500 errors).

However, if you changed the variables, you do not need to rebuild the project. You can just restart it from the main directory:
```
python start.py prod down && python start.py prod up
```

**Reply by Slavell** [reply]
Thank u very much, i found the problem, u'r advice was very helpfull.

**Comment by Slavell** [comment]
How to delete jobs?👉👈

**Reply by mlodic** [reply]
Please look at this issue I have just opened: https://github.com/intelowlproject/IntelOwl/issues/324

**Reply by eshaan7** [reply]
Just to be clear: 
An administrator can use the Django Admin Interface to delete/edit jobs however a normal user cannot delete/edit a job as of now. Tracking this in issue #324 as @mlodic mentioned.

**Comment by CITIZENDOT** [comment]
Unfortunately, You cannot. If you delete your volumes and rebuild the entire things, It will be fresh again.

`docker volume prune -a` will delete the volumes.

**Reply by mlodic** [reply]
Care because this is a destructive operation that would delete all of your users and API keys too.

**Comment by Slavell** [comment]
Hi, how can I connect by API to analyzer with self signed sertificate?

**Reply by iharshit009** [reply]
Hi Follow HTTPS with your own certificate from the [doc](https://intelowl.readthedocs.io/en/latest/Installation.html#web-server-configuration-optional). Let me know if you face any issues thereafter.
